### PR TITLE
Implemented form validation using a closured method system.

### PR DESCRIPTION
**Pull request by simonfranza**

### CODE REVIEW

**Comment by simonfranza** [comment]
[As a developer, I want to have a way to provide and trigger form field validation](https://trello.com/c/uiYkDNNj/1455-as-a-developer-i-want-to-have-a-way-to-provide-and-trigger-form-field-validation)

File: src/hooks/useForm.jsx [hunk 1]
```diff
@@ -26,14 +30,61 @@ const useForm = () => {
...
+	/**
+	 * @function
+	 * @name validateField
+	 * @description Performs a validation check on the requested input.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {string} fieldName The name of the field to perform validation on.
+	 */
+	const validateField = (fieldName) => {
```
Comment (yannhodiesne) [inline src/hooks/useForm.jsx:42]: Missing comments

File: tests/hooks/useForm.test.js [hunk 2]
```diff
@@ -187,4 +188,81 @@ describe('useForm hook', () => {
...
+			expect(hasLengthValidator).toHaveBeenCalledTimes(0);
+			expect(hasMaxLengthValidator).toHaveBeenCalledTimes(0);
+		});
+
+		it('should ignore calls with unreferenced field names', () => {
+			const dummyFieldRef = { name: 'dummy_field', rules: { isRequired: isRequiredValidator }, element: { value: '' } };
+			const { registerFormField } = sut.registerWrapper(dummyFieldRef.name);
+
+			registerFormField(dummyFieldRef);
+			sut.validateField('unreference_field');
```
Comment (yannhodiesne) [inline tests/hooks/useForm.test.js:220]: we should trigger a warning when it occurs in a dev env

File: src/validators/hasMaxLength.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @function
+ * @name hasMinLength
```
Comment (yannhodiesne) [inline src/validators/hasMaxLength.js:3]: `hasMaxLength`

**Review by yannhodiesne — APPROVED** [review]

**Pull request merged**